geth/api/backend_jail_test.go

```diff
@@ -249,9 +251,13 @@ func (s *BackendTestSuite) TestContractDeployment() {
 			from: '` + TestConfig.Account1.Address + `',
 			data: '0x6060604052341561000c57fe5b5b60a58061001b6000396000f30060606040526000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff1680636ffa1caa14603a575bfe5b3415604157fe5b60556004808035906020019091905050606b565b6040518082815260200191505060405180910390f35b60008160020290505b9190505600a165627a7a72305820ccdadd737e4ac7039963b54cee5e5afb25fa859a275252bdcf06f653155228210029',
 			gas: '` + strconv.Itoa(params.DefaultGas) + `'
-		}, function (e, contract){
+		}, function (e, contract) {
+			// NOTE: The callback will fire twice!
+			// Once the contract has the transactionHash property set and once its deployed on an address.
 			if (!e) {
-				responseValue = contract.transactionHash
+				if (!contract.address) {
+					responseValue = contract.transactionHash;
+				}
```

Comment:
Could you elaborate on this? How did it work before?

Reply:
This is probably not required anymore. This test was failing for me, but it turned out that it was a bug in the Go code. However, I left it as while debugging I found out how this callback works here https://github.com/ethereum/wiki/wiki/JavaScript-API#web3ethcontract.